src/components/periodic-table/PeriodicTable.test.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import PeriodicTable, { IPeriodicTableElement } from "./PeriodicTable";
+import {
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from "@testing-library/react";
+import PtElementInfo from "../pt-element/PtElementInfo";
+import ElementManager from "../../ElementManager";
+
+function elementRenderer(atomic: number): IPeriodicTableElement {
+  return {
+    // @ts-ignore fix this
+    component: PtElementInfo,
+    props: {
+      element: ElementManager.getElement(atomic),
+      onClick: undefined,
+    },
+  };
+}
+
+describe("should render the periodic table", () => {
+  beforeAll(() => {
+    ElementManager.loadElements();
+  });
+
+  test("should show a spinner", async () => {
+    const { container } = render(
+      <PeriodicTable elementRenderer={elementRenderer} />
+    );
+    expect(container.querySelector("svg")).toBeInTheDocument();
+  });
+
+  test("should show the periodic table", async () => {
+    const { container } = render(
+      <PeriodicTable elementRenderer={elementRenderer} />
+    );
+    await waitForElementToBeRemoved(container.querySelector("svg"));
+    expect(screen.getByText(/hydrogen/i)).toBeInTheDocument();
```

Comment:
Should we be testing that the correct number of elements(as in chemical elements) is in the document/on screen?

Reply:
Correct me if I'm mistaken, if `getAllBy` query isn't used, `getBy` query will fail due to having found multiple elements with the same criteria? :thinking: 

Or do you mean "hydrogen" has children/items? :partying_face:

Reply:
Sorry no I meant that as well as checking that hydrogen is on the page ..... rather than checking that every element renders in the periodic table.  We could make sure that the correct number of them exist instead.

Reply:
Or perhaps testing that there is the correct number of rows and columns in the table??

Reply:
@tsuki42 Take a look at what @emma-r-slight mentioned, if that is possible, checking the correct number of rows and columns, would be great I think. :100: 
Since it is the table that the user sees right? :partying_face:

Reply:
Since the data is constant, can be assume that if one of them is rendered, everything else will be too?

Reply:
> Since the data is constant

Can this test assume this though?

Reply:
Data is being loaded from a json file, so that should be constant.

Reply:
But that is an implementation detail of the component, not the test.  If it changed to load from a backend or some other data source, that assumption may change.

Reply:
I will add the tests.